### DISCUSSION TITLE
Working with shared objects .so

### DISCUSSION THREAD

**Discussion by danicc097** [jmcgeheeiv/pyfakefs#796] [Limitations]
**Describe the bug**
I ran into some testing issues with Ghostscript, which requires some shared objects to be available.

**How To Reproduce**
Install ghostscript via your system package manager and the python library `ghostscript`.

```python
class TestGhostscript:
    def test_installed(self, fs):
        from ctypes.util import find_library

        print(find_library("gs")) # expected 'libgs.so.9'

        assert find_library("gs") is not None
```

this will fail. Remove `fs` and test passes.

Is there a workaround for this? 

_NOTE_ this fails both locally and running in docker with python:3.9.5-slim-buster

**Your environment**
```bash
Linux-5.4.0-137-generic-x86_64-with-glibc2.31
Python 3.9.5 (default, Aug  3 2022, 11:09:29) 
[GCC 9.4.0]
pytest 7.1.0
pyfakefs 4.5.6
```

**Comment by mrbean-bremen** [comment]
Strange.. thanks for the report, I will see what I can find out.

**Comment by mrbean-bremen** [comment]
Sorry it so long to get to this, but I'm afraid you run into a limitation of pyfakefs here. `find_library` internally calls `subprocess.Popen` on `/sbin/ldconfig`, and `subprocess` is [known to not work](https://pytest-pyfakefs.readthedocs.io/en/latest/troubleshooting.html#subprocess-build-in) well with `pyfakefs`. It won't  even work if you map the real executable into the fake filesystem. 

So short of mocking out `find_library` to return the desired value there is nothing you can do here.

I'm going to convert this into a discussion Q/A issue so it remains visible. I added a line about `find_library` in the [documentation](https://pytest-pyfakefs.readthedocs.io/en/latest/troubleshooting.html#modules-that-rely-on-subprocess-or-multiprocessing).
